### DISCUSSION TITLE
Example-app-router error when using output: 'export' (SSG)

### DISCUSSION THREAD

**Discussion by aboutjquery** [amannn/next-intl#1262] [Q&A]
I am using example-app-router from next-intl github.

When I enter the URL without locale, for example, lcoalhost/pathname/, an error will occur. `Error: Page "/[locale]/page" is missing param "/pathnames" in "generateStaticParams()", which is required with "output : export" config.`

When the URL is added with a locale like `/en/`, it will display normally. What should I do to make it display normally without adding the locale?

If i dont use `ssg`, everything is fine.

How do I make this all work? thanks

**Comment by amannn** [comment]
Are you by chance confusing Static Site Generation and `output: export`? Those are not the same thing, you might want to review the respective docs on the Next.js side.

There's no middleware that will run when you're using `output: export`, therefore if you don't have a route defined at `/pathnames`, then this will be a 404.

I'm going to move this to a discussion since this is a usage question and not a bug report.

**Comment by ryanhex53** [comment]
When I use `headers()` in `i18n/request.ts`

```typescript
export default getRequestConfig(async () => {
    // Provide a static locale, fetch a user setting,
    // read locale from `cookies()`, `headers()`, etc.
    // const locale = LOCALES.includes(navigator.language) ? navigator.language : "en";
    // const locale = (await cookies()).get('locale')?.value || "en";
    const locale = (await headers()).get('accept-language')?.split(",")[0] || "en";

    return {
        locale,
        messages: (await import(`./messages/${locale}.json`)).default
    };
});
```

I received these errors

```
⨯ Error: Route / with `dynamic = "error"` couldn't be rendered statically because it used `headers`. See more info here: https://nextjs.org/docs/app/building-your-application/rendering/static-and-dynamic#dynamic-rendering
```

Because I have enabled output "export" in `next.config.ts`.

Do you have any recommendations if I maintain the current next.config settings and utilize the app router without i18n routing?

I would like the locale to be determined by the user's browser settings, either from request headers or navigator.language. Both methods are acceptable to me.

**Reply by sovetski** [reply]
Same question here, did you find any solution @ryanhex53 ?

**Reply by 0-don** [reply]
having the same issue

**Reply by wangdahoo** [reply]
any update of this issue?
